js_modules/dagit/packages/core/src/testing/ApolloTestProvider.tsx

```diff
@@ -22,27 +22,51 @@ export interface ApolloTestProps {
    * objects.
    */
   mocks?: any;
+
+  resolvers?: any;
 }
 
 interface Props extends ApolloTestProps {
   typeDefs: DocumentNode;
 }
 
 export const ApolloTestProvider: React.FC<Props> = (props) => {
-  const {children, mocks = [], typeDefs} = props;
+  const {children, mocks = [], typeDefs, resolvers = []} = props;
 
   const client = React.useMemo(() => {
     const mocksToMerge = Array.isArray(mocks) ? mocks : [mocks];
     const mocksWithMerge = mergeResolvers([defaultMocks, ...mocksToMerge]);
 
     const schema = makeExecutableSchema({typeDefs});
-    const mockedSchema = addMocksToSchema({schema, mocks: mocksWithMerge});
+    const mockedSchema = addMocksToSchema({
+      schema,
+      mocks: mocksWithMerge,
+      resolvers: () => {
+        const merged = mergeResolvers(resolvers);
+        const resolver = new Proxy(merged, {
```

Comment:
with this proxy we can effectively use Resolvers the same way we were using Mocks except now we get query/mutation variables passed in